src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/dates/ReleaseDateServiceTest.kt

```diff
@@ -1128,6 +1116,104 @@ class ReleaseDateServiceTest {
       }
     }
 
+    @Nested
+    inner class `Calculated PRRD licence start date` {
+      @Test
+      fun `returns null if PRRD is null`() {
+        val nomisRecord = prisonerSearchResult().copy(
+          postRecallReleaseDate = null,
+          confirmedReleaseDate = LocalDate.of(2024, 10, 22),
+        )
+
+        assertThat(service.getLicenceStartDate(nomisRecord, LicenceKind.PRRD)).isNull()
+      }
+
+      @Test
+      fun `returns PRRD if ARD is null and PRRD is a working day`() {
+        val nomisRecord = prisonerSearchResult().copy(
+          postRecallReleaseDate = LocalDate.of(2024, 10, 22),
+          confirmedReleaseDate = null,
+        )
+
+        assertThat(service.getLicenceStartDate(nomisRecord, LicenceKind.PRRD)).isEqualTo(LocalDate.of(2024, 10, 22))
```

Comment:
it'd be really nice to extract out variables to explain which date has been chosen..
e.g:
```kt
assertThat(service.getLicenceStartDate(nomisRecord, LicenceKind.PRRD)).isEqualTo(prrdDate)
```
or
```kt
assertThat(service.getLicenceStartDate(nomisRecord, LicenceKind.PRRD)).isEqualTo(nomisRecord.postRecallReleaseDate)
```

(I found some examples when trying to refactor some stuff around LSD calcuations where the test names didn't reflect the selected date)

Reply:
Done, and I've also change the date to be relative to each other, to make it easier to see `CRD is before PRRD` for example.